sbt-crossproject/src/main/scala/sbtcrossproject/CrossProject.scala

```diff
@@ -208,26 +209,38 @@ object CrossProject {
     private def sharedSrcSettings(crossType: CrossType): Seq[Setting[_]] = {
       def makeCrossSources(sharedSrcDir: Option[File],
                            scalaBinaryVersion: String,
+                           scalaEpochVersion: String,
                            cross: Boolean): Seq[File] = {
         sharedSrcDir match {
           case Some(dir) =>
             if (cross)
-              Seq(dir.getParentFile / s"${dir.name}-$scalaBinaryVersion", dir)
+              Seq(
+                dir.getParentFile / s"${dir.name}-$scalaBinaryVersion",
+                dir.getParentFile / s"${dir.name}-$scalaEpochVersion",
+                dir
+              )
             else
               Seq(dir)
           case None => Seq()
         }
       }
 
+      def scalaEpochVersion(scalaBinaryVersion: String) = {
+        val isDotty = ScalaInstance.isDotty(scalaBinaryVersion)
+        if (isDotty) "3" else "2"
+      }
+      
       Seq(
         unmanagedSourceDirectories in Compile ++= {
           makeCrossSources(crossType.sharedSrcDir(baseDirectory.value, "main"),
                            scalaBinaryVersion.value,
+                           scalaEpochVersion(scalaBinaryVersion.value),
```

Comment:
Consider calling that from `makeCrossSources` (which already has the `scalaBinaryVersion`), in order to avoid the duplicate code between Compile and Test.